include/onnxruntime/core/framework/run_options.h

```diff
@@ -17,6 +21,13 @@ struct OrtRunOptions {
   int run_log_verbosity_level = 0;  ///< VLOG level if debug build and run_log_severity_level is 0 (VERBOSE).
   std::string run_tag;              ///< A tag for the Run() calls using this.
 
+  // An execution provider specific run option is required for some execution providers, 
+  // each of which has its own resources shared within its platform.
+  unordered_map<string, void*> provider_run_options;
```

Comment:
We've seen perf issues with map lookups that have string keys and we avoid these as much as possible in the run path.

Reply:
Synced offline, this isn't a concern considering the small number of execution providers.